src/string.cpp

```diff
@@ -1197,7 +1197,7 @@ namespace bx
 
 	int32_t vsnprintf(char* _out, int32_t _max, const char* _format, va_list _argList)
 	{
-		if (1 < _max)
```

Comment:
It is a test failure before adding flush related codes in `..\bx\tests\vsnprintf_test.cpp line 31` :
```
	char buffer1[1]; // truncate
	REQUIRE(4    == bx::snprintf(buffer1, BX_COUNTOF(buffer1), "abvg") );
	REQUIRE('\0' == buffer1[BX_COUNTOF(buffer1)-1]);
```
If the require condition is what we expect, `vsnprintf` should write '\0' in case `_max == 1`.

Reply:
This test is not failing for me.

Reply:
> This test is not failing for me.

After debug, vs uses `0xcc` as uninitialized value, not 0. I think it is caused by different crt implementation.

Reply:
Fixed it: https://github.com/bkaradzic/bx/commit/85ce14956e98a37bbe52a44a645047bc98dd512d